src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,1515 @@
+  fprintf(outfile, "LSRKStep: nstsig         = %i\n", step_mem->nstsig);
+  fprintf(outfile, "LSRKStep: stagemax       = %i\n", step_mem->stagemax);
+  fprintf(outfile, "LSRKStep: stagemaxlimit  = %i\n", step_mem->stagemaxlimit);
+  fprintf(outfile, "LSRKStep: domeigfreq     = %i\n", step_mem->domeigfreq);
+
+  /* output long integer quantities */
+  fprintf(outfile, "LSRKStep: nfe            = %li\n", step_mem->nfe);
+  fprintf(outfile, "LSRKStep: domeignfe      = %li\n", step_mem->domeignfe);
+  fprintf(outfile, "LSRKStep: ndomeigupdates = %li\n", step_mem->ndomeigupdates);
+
+  /* output sunrealtype quantities */
+  fprintf(outfile, "LSRKStep: dom_eig        = %f + i%f\n", step_mem->lambdaR,
+          step_mem->lambdaI);
+  fprintf(outfile, "LSRKStep: sprad         = %f\n", step_mem->sprad);
+  fprintf(outfile, "LSRKStep: sprmax        = %f\n", step_mem->sprmax);
+  fprintf(outfile, "LSRKStep: sprmin        = %f\n", step_mem->sprmin);
+  fprintf(outfile, "LSRKStep: domeigsfty    = %f\n", step_mem->domeigsfty);
+
+  /* output sunbooleantype quantities */
+  fprintf(outfile, "LSRKStep: isextDomEig  = %d\n", step_mem->isextDomEig);
+  fprintf(outfile, "LSRKStep: newdomeig    = %d\n", step_mem->newdomeig);
+  fprintf(outfile, "LSRKStep: jacatt       = %d\n", step_mem->jacatt);
+
+#ifdef SUNDIALS_DEBUG_PRINTVEC
+  /* output vector quantities */
+
+#endif
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_Init:
+
+  This routine is called just prior to performing internal time
+  steps (after all user "set" routines have been called) from
+  within arkInitialSetup.
+
+  With initialization types FIRST_INIT this routine:
+  - sets/checks the ARK Butcher tables to be used
+  - allocates any memory that depends on the number of ARK
+    stages, method order, or solver options
+  - sets the call_fullrhs flag
+
+  With other initialization types, this routine does nothing.
+  ---------------------------------------------------------------*/
+int lsrkStep_Init(ARKodeMem ark_mem, int init_type)
+{
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeLSRKStepMem structure */
+  retval = lsrkStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* immediately return if resize or reset */
+  if (init_type == RESIZE_INIT || init_type == RESET_INIT)
+  {
+    return (ARK_SUCCESS);
+  }
+  /* enforce use of arkEwtSmallReal if using a fixed step size
+     and an internal error weight function */
+  if (ark_mem->fixedstep && !ark_mem->user_efun)
+  {
+    ark_mem->user_efun = SUNFALSE;
+    ark_mem->efun      = arkEwtSetSmallReal;
+    ark_mem->e_data    = ark_mem;
+  }
+
+  /* Set default parameters for the default stepper */
+  if (ark_mem->step == lsrkStep_TakeStepRKC)
+  {
+    step_mem->LSRKmethod   = ARKODE_LSRK_RKC_2;
+    step_mem->nfusedopvecs = 5;
+    step_mem->q = ark_mem->hadapt_mem->q = 2;
+    step_mem->p = ark_mem->hadapt_mem->p = 2;
```

Comment:
Can this be replaced with call to `LSRKStepSetMethod` to reduce code duplication and "magic numbers"